### ISSUE TITLE
[ASNetworkImageNode] Switch default image to placeholder semantics

### ISSUE THREAD

**Issue by rnystrom** [facebook/AsyncDisplayKit#347]

**Comment by lappp9** [comment]
Hey guys, so I was wanting to knock this out as a sort of [baby step](https://www.youtube.com/watch?v=lpHkFZTOIq4) to start contributing. I just have a few questions if that's ok!

I'm sure these don't really apply to this small of a change. That being said:
1.  I've never linted my Objective-C code before.  Do I just use oclint or is there something else you guys recommend?
2.  The contributing guide says that if you've changed APIs then update the documentation.  Does that just mean the 5 part getting started guide and any affected example projects?
3.  When I'm done, all I do is submit a pull request for the branch I did the work on right? I'm just paranoid that there's extra steps involved.

**Comment by wwwwwwwwwwwwwwwwwwww** [comment]
don't worry, before than always like before...

**Comment by appleguy** [comment]
Hey, great to have you in the community @lappp9!  Would love to help however I can.

1. Embarrassing, but I'm actually not sure how the linter is set up for ASDK if we have one.  If there's an oclint configuration in the project, run it!  I haven't been doing this on my changes, lol.
2. I wouldn't worry about updating the docs as a blocker for this, but yes, if you can update the gh-pages branch with any relevant information then that would be really cool.  There are also generated doc pages (accessible from some links on asyncdisplaykit.org) but those are created by running a script on the codebase, scraped from the specific comment format that we use in header files.  We are due to re-generate them now, actually (although it's possible @secretiverhyme set up a magical automatic updating script :) )
3. Yes, just submit a pull request that diffs between your forked project's branch and the facebook/AsyncDisplayKit master branch.

**Comment by lappp9** [comment]
Ok awesome, thanks for the tips @appleguy!  I'll definitely keep that in mind for my pull requests moving forward.  

Oh, one last question:
As far as testing goes, I noticed there aren't any tests for ASNetworkImageNode but there were ASMultiplexImageNode tests around more complicated functionality.  Is the idea that you only put tests in for stuff that seems like it really needs it?  That's what it seems like, but I figured I might as well ask to make sure.

**Comment by appleguy** [comment]
@lappp9 Test coverage in ASDK is acceptable, but definitely incomplete.  To be honest, it's tough to have a test that hurts — even simple tests are nice to have, as long as they are not redundant.  For example, you would not want to test basic properties of ASDisplayNode on ASNetworkImageNode, like verifying that .bounds is the same as what you set it to, but checking features of ASNetworkImageNode itself could be useful.

**Comment by lappp9** [comment]
Ok that definitely makes sense.  I'll throw a few simple tests in for ASNetworkImageNode then!

**Comment by lappp9** [comment]
@appleguy Ok so I tried to add two simple tests for ASNetworkImageNode and when I open a pull request the travis build fails on the ASNetworkImageNode import statement.  Is there some kind of extra configuration I need to do for any test classes I add?  

I just figured I would ask in case this was something obvious to you.  If not, I can just keep digging to see what the problem is.

As a side question, I also looked through some other tests to get a feel for them and noticed that a few classes set the -fno-objc-arc compiler flag and I was wondering why that was.
